src/test/java/emissary/test/core/junit5/RegressionTest.java

```diff
@@ -141,47 +143,52 @@ protected ElementEncoders getEncoders() {
             final String tname) {
+
         if (!IBaseDataObjectXmlCodecs.SHA256_ELEMENT_ENCODERS.equals(getEncoders())) {
             return;
         }
 
+        // touch up alternate views to match how their bytes would have encoded into the answer file
         for (Entry<String, byte[]> entry : new TreeMap<>(payload.getAlternateViews()).entrySet()) {
-            payload.addAlternateView(entry.getKey(), ByteUtil.sha256Bytes(entry.getValue()).getBytes(StandardCharsets.ISO_8859_1));
+            Optional<String> viewSha256 = hashBytesIfNonPrintable(entry.getValue());
+            viewSha256.ifPresent(s -> payload.addAlternateView(entry.getKey(), s.getBytes(StandardCharsets.ISO_8859_1)));
         }
 
-        if (payload.data() != null && ByteUtil.hasNonPrintableValues(payload.data())) {
-            final String hash = ByteUtil.sha256Bytes(payload.data());
-
-            if (hash != null) {
-                payload.setData(hash.getBytes(StandardCharsets.UTF_8));
-            }
-        }
+        // touch up primary view if necessary
+        Optional<String> payloadSha256 = hashBytesIfNonPrintable(payload.data());
+        payloadSha256.ifPresent(s -> payload.setData(s.getBytes(StandardCharsets.UTF_8)));
 
         if (payload.getExtractedRecords() != null) {
             for (final IBaseDataObject extractedRecord : payload.getExtractedRecords()) {
-                if (ByteUtil.hasNonPrintableValues(extractedRecord.data())) {
-                    final String hash = ByteUtil.sha256Bytes(extractedRecord.data());
-
-                    if (hash != null) {
-                        extractedRecord.setData(hash.getBytes(StandardCharsets.UTF_8));
-                    }
-                }
+                Optional<String> recordSha256 = hashBytesIfNonPrintable(extractedRecord.data());
+                recordSha256.ifPresent(s -> extractedRecord.setData(s.getBytes(StandardCharsets.UTF_8)));
             }
         }
 
         if (attachments != null) {
             for (final IBaseDataObject attachment : attachments) {
                 if (ByteUtil.hasNonPrintableValues(attachment.data())) {
-                    final String hash = ByteUtil.sha256Bytes(attachment.data());
-
-                    if (hash != null) {
-                        attachment.setData(hash.getBytes(StandardCharsets.UTF_8));
-                    }
+                    Optional<String> attachmentSha256 = hashBytesIfNonPrintable(attachment.data());
+                    attachmentSha256.ifPresent(s -> attachment.setData(s.getBytes(StandardCharsets.UTF_8)));
                 }
             }
         }
     }
 
+    /**
+     * Generates a SHA 256 hash of the provided bytes if they contain any non-printable characters
+     * 
+     * @param bytes the bytes to evaluate
+     * @return a value optionally containing the generated hash
+     */
+    private Optional<String> hashBytesIfNonPrintable(byte[] bytes) {
```

Comment:
Should this be part of RegressionTestUtil?

Reply:
Good with this change or not.